### ISSUE TITLE
Support for user input

### ISSUE THREAD

**Issue by okonet** [SamVerschueren/listr#32]
Sometimes tasks can require some user input from the terminal. Right now, it doesn't seem to be possible to pause the render that will always eat the user input. I'm not sure if this is even possible with a custom renderer. Either way, I think user input should be supported.

Can you point me in a right direction here?

**Comment by okonet** [comment]
Okay, seems it should be possible with a renderer. It is still working weird right now. I'll investigate and post more here.

**Comment by SamVerschueren** [comment]
A task could be in a pending state as long as no user input is provided, no? Do you have something like a custom renderer start that you could share? And maybe also a quick listr task that asks for input so I can investigate with you. Either a gist or a repository would be fine.

Thanks for looking into this @okonet !

**Comment by SamVerschueren** [comment]
@okonet I tried to ask for input with the verbose renderer, seems to work fine.

```js
'use strict';
const Listr = require('listr');
const promptly = require('promptly');

const tasks = new Listr([
	{
		title: 'Hello',
		task: () => {
			return promptly.prompt('Name: ')
				.then(name => {
					console.log(name);
				});
		}
	}
]);

tasks.run();
```

**Comment by okonet** [comment]
Yes I was looking to have a promt in the default renderer. I still didn't have time to look into this :(

**Comment by okonet** [comment]
Another thing I've noticed is that even with the verbose renderer it got skipped if run in a pre-commit hook. It worked when running through `npm run` though. I'll need to investiagte that.

**Comment by SamVerschueren** [comment]
@okonet No rush! I don't expect anything from my contributors. I'm already very glad other people try to think with me :).

**Comment by SamVerschueren** [comment]
Hi @okonet, because we needed input for two factor authentication in `np`, I decided to dive deeper and created [listr-input](https://github.com/SamVerschueren/listr-input). It's not ideal yet and it can definitely be improved, but it's a starting point. Feel free to give it a go and provide feedback!

Here's the PR where I added it to `np`, might give an idea of how it can look https://github.com/sindresorhus/np/pull/200.

**Comment by okonet** [comment]
@SamVerschueren that's amazing! Thanks for letting me know!

**Comment by aequasi** [comment]
Would be nice to see added support for any prompting library. Oclif suggests the use of inquirer (along with another lib) and I'd love to use it in Listr.
